src/components/Global/BaseCheckbox.vue

```diff
@@ -0,0 +1,79 @@
+<script>
+import { defineComponent } from "vue";
+
+export default defineComponent({
+  name: "BaseCheckbox",
+  props: {
+    modelValue: {
+      type: Boolean,
+    },
+    name: {
+      type: String,
+    },
+    label: {
+      type: String,
+    },
+  },
+  emits: ["update:modelValue"],
+  methods: {
+    updateValue(event) {
+      this.$emit("update:modelValue", event.target.checked);
+    },
+  },
+});
+</script>
+
+<template>
+  <div class="input__wrapper">
+    <label class="input__label" :class="$attrs.class" :for="name">
+      <span class="input__icon" :class="$attrs.class"></span>
+      <input
+        class="input__checkbox"
+        :class="$attrs.class"
+        type="checkbox"
+        :id="name"
+        :name="name"
+        :value="modelValue"
+        @input="updateValue"
+      />{{ label }}</label
+    ><slot />
+  </div>
+</template>
+
+<style lang="scss" scoped>
+.input__wrapper {
+  padding-top: 25px;
+  display: flex;
+  align-items: center;
+  gap: 6px;
+}
+.input__label {
+  font-weight: 400;
+  font-size: 1.1rem;
+  line-height: 1.1rem;
+  color: $colorGreyCharade;
+  cursor: pointer;
+  display: flex;
+  align-items: center;
+  gap: 12px;
+}
+.input__icon {
+  width: 24px;
+  height: 24px;
+  border: 1px solid $colorGreyBombay;
+  border-radius: 9px;
+  cursor: pointer;
+  &.checked {
+    background-image: url("@/assets/images/checked.svg");
+    background-size: contain;
+    border: 1px solid $colorRedBittersweet;
+  }
+}
+.input__checkbox {
+  position: absolute;
+  opacity: 0;
```

Comment:
one issue with this approach is that the checkbox won't be focusable or accessible with keyboard navigation.

For some inspiration here's a great article about how to style checkboxes:
https://moderncss.dev/pure-css-custom-checkbox-style/

Reply:
Thank you, I was told the appearance is not safe enough, but it could be outdated info. I corrected styling so now it is accessible with keyboard.